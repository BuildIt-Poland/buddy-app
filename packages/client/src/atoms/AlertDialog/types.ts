export interface AlertDialogProps {
  isOpen: boolean;
  title?: string;
  message: string;
  onConfirm?:
    | ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)
    | undefined;
  onClose?:
    | ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)
    | undefined;
}
