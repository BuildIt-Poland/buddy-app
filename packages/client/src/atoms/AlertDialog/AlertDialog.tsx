import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { AlertDialogProps } from './types';

const AlertDialog: React.FC<AlertDialogProps> = ({
  isOpen,
  title,
  message,
  onConfirm,
  onClose,
}) => {
  return (
    <Dialog open={isOpen} onClose={onClose} data-testid='alert-dialog'>
      {title && (
        <DialogTitle>
          <strong>{title}</strong>
        </DialogTitle>
      )}
      <DialogContent>
        <DialogContentText variant={'body2'} style={{ whiteSpace: 'pre-line' }}>
          {message}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose} color='primary' data-testid='alert-dialog-close'>
          Close
        </Button>
        {onConfirm && (
          <Button
            onClick={onConfirm}
            color='primary'
            data-testid='alert-dialog-confirm'>
            Confirm
          </Button>
        )}
      </DialogActions>
    </Dialog>
  );
};

export default AlertDialog;
