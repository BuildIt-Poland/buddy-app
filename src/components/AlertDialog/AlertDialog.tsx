import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

interface AlertDialogProps {
  message?: string;
  title?: string;
}

const AlertDialog = ({ title, message }: AlertDialogProps) => {
  const [open, setOpen] = React.useState(true);

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <Dialog open={open} onClose={handleClose} data-testid='alert-dialog'>
      {title && <DialogTitle>{title}</DialogTitle>}
      <DialogContent>
        {message && (
          <DialogContentText variant={'body2'} style={{ whiteSpace: 'pre-line' }}>
            {message}
          </DialogContentText>
        )}
      </DialogContent>
      <DialogActions>
        <Button
          onClick={handleClose}
          color='primary'
          data-testid='alert-dialog-close'>
          Close
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default AlertDialog;
