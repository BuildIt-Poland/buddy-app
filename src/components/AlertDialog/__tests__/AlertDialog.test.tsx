import React from 'react';
import { create } from 'react-test-renderer';
import AlertDialog from '../AlertDialog';

jest.mock('@material-ui/core/Button', () => 'Button');
jest.mock('@material-ui/core/Dialog', () => 'Dialog');
jest.mock('@material-ui/core/DialogActions', () => 'DialogActions');
jest.mock('@material-ui/core/DialogContent', () => 'DialogContent');
jest.mock('@material-ui/core/DialogContentText', () => 'DialogContentText');
jest.mock('@material-ui/core/DialogTitle', () => 'DialogTitle');

describe('Component - AlertDialog', () => {
  test('renders correctly', () => {
    const component = create(<AlertDialog />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
