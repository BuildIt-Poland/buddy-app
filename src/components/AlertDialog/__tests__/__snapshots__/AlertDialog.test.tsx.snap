// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - SnackBar renders correctly 1`] = `
<Dialog
  data-testid="alert-dialog"
  onClose={[Function]}
  open={false}
>
  
  <DialogContent>
    <DialogContentText
      style={
        Object {
          "whiteSpace": "pre-line",
        }
      }
      variant="body2"
    >
      
    </DialogContentText>
  </DialogContent>
  <DialogActions>
    <Button
      color="primary"
      data-testid="alert-dialog-close"
      onClick={[Function]}
    >
      Close
    </Button>
    <Button
      color="primary"
      data-testid="alert-dialog-confirm"
      onClick={[Function]}
    >
      Confirm
    </Button>
  </DialogActions>
</Dialog>
`;
